src/displayapp/screens/settings/SettingPineTimeStyle.cpp

```diff
@@ -0,0 +1,332 @@
+      lv_obj_set_style_local_text_color(timeAMPM, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueTime]);
+    }
+  }
+  if((object == btnPrevTime) && (event == LV_EVENT_PRESSED)) {
+    if ( valueTime > 0 ) {
+      valueTime -= 1;
+      settingsController.SetPTSColorTime(valueTime);
+      lv_obj_set_style_local_text_color(timeDD1, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueTime]);
+      lv_obj_set_style_local_text_color(timeDD2, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueTime]);
+      lv_obj_set_style_local_text_color(timeAMPM, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueTime]);
+    }  else {
+      valueTime = 16;
+      settingsController.SetPTSColorTime(valueTime);
+      lv_obj_set_style_local_text_color(timeDD1, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueTime]);
+      lv_obj_set_style_local_text_color(timeDD2, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueTime]);
+      lv_obj_set_style_local_text_color(timeAMPM, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueTime]);
+    }
+  }
+  if((object == btnNextBar) && (event == LV_EVENT_PRESSED)) {
+    if ( valueBar < 16 ) {
+      valueBar += 1;
+      settingsController.SetPTSColorBar(valueBar);
+      lv_obj_set_style_local_bg_color(sidebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBar]);
+    } else {
+      valueBar = 0;
+      settingsController.SetPTSColorBar(valueBar);
+      lv_obj_set_style_local_bg_color(sidebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBar]);
+    }
+  }
+  if((object == btnPrevBar) && (event == LV_EVENT_PRESSED)) {
+    if ( valueBar > 0 ) {
+      valueBar -= 1;
+      settingsController.SetPTSColorBar(valueBar);
+      lv_obj_set_style_local_bg_color(sidebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBar]);
+    } else {
+      valueBar = 16;
+      settingsController.SetPTSColorBar(valueBar);
+      lv_obj_set_style_local_bg_color(sidebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBar]);
+    }
+  }
+  if((object == btnNextBG) && (event == LV_EVENT_PRESSED)) {
+    if ( valueBG < 16 ) {
+      valueBG += 1;
+      settingsController.SetPTSColorBG(valueBG);
+      lv_obj_set_style_local_bg_color(timebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBG]);
+    } else {
+      valueBG = 0;
+      settingsController.SetPTSColorBG(valueBG);
+      lv_obj_set_style_local_bg_color(timebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBG]);
+    }
+  }
+  if((object == btnPrevBG) && (event == LV_EVENT_PRESSED)) {
+    if ( valueBG > 0 ) {
+      valueBG -= 1;
+      settingsController.SetPTSColorBG(valueBG);
+      lv_obj_set_style_local_bg_color(timebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBG]);
+    } else {
+      valueBG = 16;
+      settingsController.SetPTSColorBG(valueBG);
+      lv_obj_set_style_local_bg_color(timebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBG]);
+    }
+  }
```

Comment:
In each of these inner if statements some of the same code is duplicated in both `if` and `else`. It could be moved outside of it.

Also `else if` can be used on the object checks.

Reply:
Thanks, resolved.